scripts/utils/splitLandIdsForTunnelMigration.ts

```diff
@@ -0,0 +1,236 @@
+  fs.readFileSync('./tunnel-polygon.json').toString()
+);
+
+const GRID_SIZE = 408;
+
+interface TokenData {
+  id: string;
+  owner: {
+    id: string;
+  };
+}
+
+interface Quad {
+  size: number;
+  x: number;
+  y: number;
+}
+
+const quads3x3OnLayer1: Array<Quad> = [];
+const quads6x6OnLayer1: Array<Quad> = [];
+const quads12x12OnLayer1: Array<Quad> = [];
+const quads24x24OnLayer1: Array<Quad> = [];
+const quads3x3OnLayer2: Array<Quad> = [];
+const quads6x6OnLayer2: Array<Quad> = [];
+const quads12x12OnLayer2: Array<Quad> = [];
+const quads24x24OnLayer2: Array<Quad> = [];
+
+let tokenOnLayer1: Array<number> = [];
+let remainingTokenOnL2: Array<number> = [];
+
+for (let i = 0; i < tokensSnapshotL1.length; i++) {
+  tokenOnLayer1[i] = parseInt(tokensSnapshotL1[i].id);
+}
+
+// fetch common IDs
+function getCommonElement(arr1: Array<TokenData>, arr2: Array<TokenData>) {
+  let pointer = 0;
+  const output = [];
+  for (let i = 0; i < arr1.length; ++i) {
+    for (let j = pointer; j < arr2.length; ++j) {
+      if (arr1[i].id == arr2[j].id) {
+        pointer = j;
+        output.push(parseInt(arr1[i].id));
+      }
+    }
+  }
+  return output;
+}
+
+const commonIds = getCommonElement(tokensSnapshotL1, tokensSnapshotL2);
+
+for (let i = 0; i < tokensSnapshotL2.length; i++) {
+  remainingTokenOnL2.push(parseInt(tokensSnapshotL2[i].id));
+}
+for (let i = 0; i < commonIds.length; i++) {
+  remainingTokenOnL2 = remainingTokenOnL2.filter(
+    (element: number) => element != commonIds[i]
+  );
+}
+
+function getParentQuadCoordinates(tokenId: number, parentSize: number) {
+  const x = Math.floor((tokenId % GRID_SIZE) / parentSize) * parentSize;
+  const y =
+    Math.floor(Math.floor(tokenId / GRID_SIZE) / parentSize) * parentSize;
+  return {x, y};
+}
+
+function idInPath(i: number, size: number, x: number, y: number) {
+  const row = Math.floor(i / size);
+  if (row % 2 == 0) {
+    return x + (i % size) + (y + row) * GRID_SIZE;
+  } else {
+    return x + size - (1 + (i % size)) + (y + row) * GRID_SIZE;
+  }
+}
+
+function checkOwner(size: number, x: number, y: number, arr: Array<number>) {
+  let flag = true;
+  for (let i = 0; i < size * size; i++) {
+    const id = idInPath(i, size, x, y);
+    flag = flag && arr.includes(id);
+  }
+  return flag;
+}
+
+function checkOwnerAndReturnQuad(tokenId: number, arr: Array<number>) {
+  let coordinates = getParentQuadCoordinates(tokenId, 24);
+  let isOwner = checkOwner(24, coordinates.x, coordinates.y, arr);
+  let size = 24;
+  if (!isOwner) {
+    coordinates = getParentQuadCoordinates(tokenId, 12);
+    isOwner = checkOwner(12, coordinates.x, coordinates.y, arr);
+    size = 12;
+  }
+  if (!isOwner) {
+    coordinates = getParentQuadCoordinates(tokenId, 6);
+    isOwner = checkOwner(6, coordinates.x, coordinates.y, arr);
+    size = 6;
+  }
+  if (!isOwner) {
+    coordinates = getParentQuadCoordinates(tokenId, 3);
+    isOwner = checkOwner(3, coordinates.x, coordinates.y, arr);
+    size = 3;
+  }
```

Comment:
We need to check for individual quads that could be formed around that 1x1 land.
They are of 4 Types and should be checked individually for each case as in case of recursion there is no single point to call check all the cases and get the exact quad that could be formed.

Reply:
I'm not sure i understand so i tried to code what I had in mind and it should be identical to your code

```
  let isOwner = checkOwner(24, coordinates.x, coordinates.y, arr);
```

```
function checkOwner(tokenId:number, size: number, x: number, y: number, arr: Array<number>) {
  let flag = true;
  for (let i = 0; i < size * size; i++) {
    const id = idInPath(i, size, x, y);
    flag = flag && arr.includes(id);
  }
 
  if (!flag) {
    coordinates = getParentQuadCoordinates(tokenId, size / 2);
    return checkOwner(size / 2, coordinates.x, coordinates.y, arr);  
  }

  return flag;
}
```

Reply:
Got it, it just needs to return the co-ordinates as  well.